selendroid-server/src/main/java/io/selendroid/server/model/DefaultAppiumUiAutomatorDriver.java

```diff
@@ -0,0 +1,279 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.selendroid.server.model;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.*;
+import io.selendroid.server.ServerInstrumentation;
+import io.selendroid.server.android.InstrumentedKeySender;
+import io.selendroid.server.android.KeySender;
+import io.selendroid.server.android.internal.Dimension;
+import io.selendroid.server.common.exceptions.NoSuchMethodImplementationException;
+import io.selendroid.server.common.exceptions.SelendroidException;
+import io.selendroid.server.util.SelendroidLogger;
+
+
+public class DefaultAppiumUiAutomatorDriver implements SelendroidDriver {
+
+    public static final String BROWSER_NAME = "browserName";
+    public static final String PLATFORM = "platform";
+    public static final String SUPPORTS_JAVASCRIPT = "javascriptEnabled";
+    public static final String TAKES_SCREENSHOT = "takesScreenshot";
+    public static final String SUPPORTS_ALERTS = "handlesAlerts";
+    public static final String ROTATABLE = "rotatable";
+    public static final String ACCEPT_SSL_CERTS = "acceptSslCerts";
+    public static final String SUPPORTS_NETWORK_CONNECTION = "networkConnectionEnabled";
+    public static final String AUTOMATION_NAME = "automationName";
+    private ServerInstrumentation serverInstrumentation = null;
+    private KeySender keySender = null;
+    private Session session = null;
+
+    public DefaultAppiumUiAutomatorDriver(ServerInstrumentation serverInstrumentation) {
+        this.keySender = new InstrumentedKeySender(serverInstrumentation);
+        this.serverInstrumentation = serverInstrumentation;
+    }
+
+    @Override
+    public AndroidElement findElement(By by) {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public List<AndroidElement> findElements(By by) {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public String getCurrentUrl() {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public Session getSession() {
+        return session;
+    }
+
+    @Override
+    public JSONObject getSessionCapabilities(String sessionId) {
+        SelendroidLogger.info("session: " + sessionId);
+
+        JSONObject copy;
+        try {
+            JSONObject capabilities = session.getCapabilities();
+            if (capabilities != null) {
+                copy = new JSONObject(capabilities.toString());
+            } else {
+                copy = new JSONObject();
+            }
+            copy.put(TAKES_SCREENSHOT, true);
+            copy.put(BROWSER_NAME, "selendroid");
```

Comment:
The default capabilities will be set only if the session does not contain any capabilities.
```
     if (capabilities != null) {
                copy = new JSONObject(capabilities.toString());
            } else {
                copy = new JSONObject();
            }
```
and yes the var name `copy` needs to changed.